app/views/sessions/purchase_cart.html.erb

```diff
@@ -7,7 +7,11 @@
     </tr>
     <% if session[:cart] %>
         <% session[:cart].each do |item_id, quantity| %>
-        <% item = Item.find_by(id: item_id) %>
+            <% item = Item.find_by(id: item_id) %>
+            <!-- Line below generates receipt for purchasing this item, if possible -->
+            <% if logged_in? %>
```

Comment:
Why not just have them log in to view their receipts? It doesn't take much more effort from the user and is arguably easier than typing in a 10 digit code. Again, it would be weird if I could go to some random URL and creep on your EBay and Amazon purchases, right?